xapian-core/queryparser/mathtermgenerator_internal.cc

```diff
@@ -350,16 +352,16 @@ MathTermGenerator::Internal::index_math(const char * ch, const string & prefix)
 
 const unsigned MAX_PATH_LEN = 3;
 
-void
+vector<string>
 MathTermGenerator::Internal::generate_symbol_pair_list()
 {
     // If main row has size n, then number of tuples along main row is
     // T = (n * (n+1))/ 2. if main row has trow and brow of each size 2, then
     // total pairs is T + 4 * T. This computation is not accurate, an attempt to
     // avoid reallocation.
     unsigned long approx_pairs_count = (mrow.size() * (mrow.size() + 1)) * 3;
```

Comment:
`unsigned long` is the wrong type here really - the `size()` methods will return `std::vector<Foo>::size_type` but that should be the same as `size_t`.  So `size_t` or just `auto` is a better choice.